src/yoroi-wallets/cardano/api.ts

```diff
@@ -184,3 +211,31 @@ export const fetchCurrentPrice = async (currency: CurrencySymbol, config: Backen
 
   return response.ticker.prices[currency]
 }
+
+export const parseModerationStatus = (status: unknown): YoroiNFTModerationStatus | undefined => {
+  const statusString = String(status)
+  const map = {
+    RED: 'blocked',
+    YELLOW: 'consent',
+    GREEN: 'approved',
+    PENDING: 'pending',
+    MANUAL_REVIEW: 'manual_review',
+  } as const
+  return map[statusString.toUpperCase() as keyof typeof map]
+}
+
+function parseNFTs(value: unknown, storageUrl: string): YoroiNFT[] {
+  if (!value || typeof value !== 'object') {
+    return []
+  }
+
+  const assets = Object.values(value).map((arr) => arr[0])
+  const nftAssets: NFTAsset[] = assets.filter((asset) => isAssetNFT(asset))
+  return nftAssets.map((nft) => convertNft(nft.metadata, storageUrl))
+}
+
+function isAssetNFT(asset: unknown): asset is NFTAsset {
+  return typeof asset === 'object' && !!asset && (asset as NFTAsset).key === NFT_METADATA_KEY
```

Comment:
```
  return isObject(asset) && 'key' in asset && asset.key === NFT_METADATA_KEY
```
add isObject to `parsing.ts`

Reply:
`asset.key === NFT_METADATA_KEY` won't work, because `Property 'key' does not exist on type 'object'`. I'll create the helper util, but we should really look into some library to use for parsing the data

Reply:
What do you think about this @wolverineks 
```

export const isObject = (data: unknown): data is object => {
  return typeof data === 'object' && data !== null && !Array.isArray(data)
}

export const hasProperties = <T extends object, K extends string>(
  obj: T,
  ...keys: K[]
): obj is T & {[J in K]: unknown} => {
  return !!obj && keys.every((key) => Object.prototype.hasOwnProperty.call(obj, key))
}
```

Reply:
Then we can do

```
function isAssetNFT(asset: unknown): asset is NFTAsset {
  return isObject(asset) && hasProperties(asset, 'key') && asset.key === NFT_METADATA_KEY
}
```